### DISCUSSION TITLE
How to use preloaded ChromeDriver?

### DISCUSSION THREAD

**Discussion by blexey** [WikiTransformationProject/wikitraccs-releases#95] [Q&A] [answered]
My configuration:
- WikiTraccs 1.8.6 from WikiTraccs_1.8.6_windows_x64_release_selfcontained.zip unpacked right to the Desktop
- Google Chrome version 119.0.6045.106 (Official Build) (64-bit)
- No network access to some external web resources including https://chromedriver.storage.googleapis.com
- Confluence Server site accessible in Chrome

WikiTraccs.GUI.exe complains that it unable to download ChromeDriver by itself, as described [here](https://www.wikitransformationproject.com/docs/troubleshooting/connectivity-issues/)

How it is possible to point WikiTraccs to already downloaded chromedriver.exe?
[This](https://www.wikitransformationproject.com/docs/reference/settings/configuration-file/) article suggests to use custom appsettings.json, but the most suitable option WebDriverDirPath means the place to which WebDriver is going to be downloaded by WikiTraccs anyway. Is it possible to skip the download phase and use preloaded one?

**Comment by heinrich-ulbricht** [comment]
@blexey Thanks for the detailed question!

Please have a look here at this section: https://www.wikitransformationproject.com/docs/troubleshooting/connectivity-issues/#how-to-handle-blocked-google-chrome-webdriver-endpoints

This aims to explain how to get the chromedriver.exe. You would download it manually on another, unblocked machine, then copy the driver over to the machine WikiTraccs is running on into the folder specified by `WebDriverDirPath`. `ChromeDriverVersionOverride` needs to be filled as well.

WikiTraccs will still complain about blocked endpoints, when doing the connection check. But it should pick up the manually provided driver and open the browser to log into Confluence.

Let me know if you get stuck.

**Comment by blexey** [comment]
@heinrich-ulbricht Unfortunately, it doesn't work.
Screenshots attached:
![Screenshot 2023-11-06 125931](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/2093853/7b74ee5c-8a4b-4d6e-b50d-761249eea7eb)
![Screenshot 2023-11-06 130423](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/2093853/347cadff-64dd-4baf-8942-9816842a157e)

**Reply by heinrich-ulbricht** [reply]
Now that does not look right. Will check.

**Comment by blexey** [comment]
> Now that does not look right. Will check.
As I've found, this is an issue with system settings or policies, not with WikiTraccs, because it works perfectly on my local machine.

**Reply by heinrich-ulbricht** [reply]
@blexey Interesting. Would you mind sending the [common log files](https://www.wikitransformationproject.com/docs/troubleshooting/troubleshooting-strategies/#common-log-files) of such a failed attempt to support @ wikitransformationproject.com ? I'd like to see place where the error occurs. The error message in WikiTraccs should point in a better direction. Right now it doesn't make sense from a user perspective.

**Comment by blexey** [comment, marked as answer]
Thank you so much! I've managed to run Chrome by WikiTraccs and successfully connected to my endpoints.
Actually, the issue was in mismatching versions of ChromeDriver and Chrome.
